polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -324,6 +327,25 @@ impl CandidateCommitments {
 			UMPSignal::SelectCore(core_selector, cq_offset) => Some((core_selector, cq_offset)),
 		}
 	}
+
+	/// Returns the core index determined by `UMPSignal::SelectCore` commitment
+	/// and `assigned_cores`.
+	///
+	/// Returns `None` if there is no `UMPSignal::SelectCore` commitment or
+	/// assigned cores is empty.
+	///
+	/// `assigned_cores` must be a sorted vec of all core indices assigned to a parachain.
+	pub fn committed_core_index(&self, assigned_cores: &[&CoreIndex]) -> Option<CoreIndex> {
+		if assigned_cores.is_empty() {
+			return None
+		}
+
+		self.selected_core().and_then(|(core_selector, _cq_offset)| {
+			let core_index =
+				**assigned_cores.get(core_selector.0 as usize % assigned_cores.len())?;
```

Comment:
what i didn't get from the RFC is why do we need to do `% assigned_cores` and not assume that the `core_selector < assigned_cores.len()`?

Reply:
Core selector is a sequence number that wraps around. We use the LSB of block numbers to fill it in.

Reply:
yes, and it we need it to wrap around because the core selector is a u8